derp/derphttp/derphttp_client.go

```diff
@@ -234,13 +234,27 @@ func (c *Client) connect(ctx context.Context, caller string) (client *derp.Clien
 	return c.client, nil
 }
 
+func reconnectRequired(err error) bool {
```

Comment:
derp.Client methods could also return something like `ErrUnusableTransport` to indicate cases that require reconnecting (as opposed to cases when the current transport remains usable). Just raw networking errors are insufficiet.